socialhome/streams/app/stores/streamStore.js

```diff
@@ -10,15 +11,65 @@ import {actions, mutations, streamStoreOperations, getters} from "streams/app/st
 
 Vue.use(Vuex)
 
+function newRestAPI(options) {
+    const opts = _defaults({}, options, {
+        baseURL: "/api",
+        axios: Vue.prototype.$http,
+    })
+
+    const onSuccess = (state, payload) => {
+        payload.data.forEach(item => {
+            Vue.set(state.contents, item.id, item)
+            state.contentIds.push(item.id)
+        })
+    }
+
+    const onError = (state, error) => {
+        /* TODO: Proper error handling */
+        console.error(`An error happened while fetching post: ${error}`)
+    }
+
+    return new Vapi(opts)
+        .get({
+            action: "getPublicStream",
+            path: "/streams/public",
+            property: "contents",
+            onSuccess,
+            onError,
+        })
+        .get({
+            action: "getFollowedStream",
+            path: "/streams/followed",
+            property: "contents",
+            onSuccess,
+            onError,
+        })
+        .get({
+            action: "getTagStream",
+            path: ({name}) => `/streams/tag/${name}`,
+            property: "contents",
+            onSuccess,
+            onError,
+        })
+        .get({
+            action: "getProfileStream",
+            path: ({id}) => `/profiles/${id}`,
```

Comment:
This is actually not the right url for profile content stream, it would likely be `api:streams-profile` with a parameter, and `api:streams-profile-all` with parameter for that version. But, I have to make it first. So I forgive you ;)